### DISCUSSION TITLE
Network v19 Lightning and v20 Thunder Upgrade and Lotus Releases Timeline

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#10686] [Announcement]
**More details on the Lightning and Thunder network upgrade can be found [here](https://github.com/filecoin-project/community/discussions/74#discussioncomment-5376097)!**

Subscribe to notifications on https://status.filecoin.io/ for any additional updates!
If you have any concerns or questions, leave a comment in #fil-lotus-help.

## Lotus releases 
**Note: This projected timeline may be subject to change based on testing and development!**

You will need to be on lotus v1.22.0 or later for the Lightning ⚡️ and Thunder ⛈ upgrade. We will be publishing two lotus releases before this upgrade:

- **[lotus `v1.22.0`]** It is based on the last stable release, lotus `v1.20.4`, and includes **only the consensus-breaking changes** for the Network nv19 & nv20 upgrade. We recommend ecosystem teams like node operators to upgrade your node to this version as soon as the release is tagged.
- [**lotus `v1.23.0`**] This Lotus release will include everything in `v1.22.0` plus the latest functionality and improvements we have been [testing in v1.21.0](https://github.com/filecoin-project/lotus/releases/tag/v1.21.0-rc3) for the last 6 weeks. That being said, with the new timeline of nv19 and its corresponding release v1.22.0, we will bump v1.21.0 to v1.23.0 - so it includes all codes for **network upgrade** AND all the tested new features and improvements. **We highly recommend users like node service providers, exchanges, splitstore users to upgrade your nodes and all systems to this release once it's out!**

> ⚠️ Go version: Lotus v1.22.0 will NOT be compatible with go v1.20.X. While Lotus v1.23.0 will be.

## Tentative Releases and Network Upgrade Timelines

> Time below is in UTC.
- [x] 2023-04-17, [lotus v1.22.0-rc4](https://github.com/filecoin-project/lotus/releases/tag/v1.22.0-rc4) (Note: This release candidate sets the upgrade epochs for calibration, and does not set the upgrade epochs for mainnet.)
- [x] 2023-04-19 [lotus v1.23.0-rc1](https://github.com/filecoin-project/lotus/releases/tag/v1.22.0-rc6)(Note: This release candidate sets the upgrade epochs for calibration, and does not set the upgrade epochs for mainnet.)
- [x] 2023-04-19, [lotus v1.22.0-rc6](https://github.com/filecoin-project/lotus/releases/tag/v1.22.0-rc6) (Note: This release candidate sets the upgrade epochs for calibration, and does not set the upgrade epochs for mainnet.)
- [x] 2023-04-20T14:00:00Z at epoch `489094`, Calibnet Lightning ⚡️ nv19 upgrade 
- [x] 2023-04-21T16:00:00Z at epoch `492214`, Calibnet Thunder ⛈ nv19 upgrade 
- [x] 2023-04-21, [lotus v1.22.1](https://github.com/filecoin-project/lotus/releases/tag/v1.22.1) (Note: The final release that sets the upgrade epochs for mainnet.)
- [x] 2023-04-24, [lotus v1.23.0](https://github.com/filecoin-project/lotus/releases/tag/v1.23.0) (Note: The final release that sets the upgrade epochs for mainnet.)
- [x] 2023-04-27T13:00:00Z Mainnet lightning upgrade ⚡️ at epoch `2809800`!
- [ ] 2023-05-18T13:00:00Z Mainnet Thunder upgrade ⚡️ at epoch `2870280`!

**Comment by jennijuju** [comment]
## Releases are out!

You need to be on lotus v1.22.1 or up for this upgrade. Both v1.22.1 and v1.23.0 are production ready releases and you may chose the one that's better for you.

🤔  Wondering which Lotus version you should update to for the coming network upgrade, here is the TL;DR:
- If you are on a Lotus v1.20.x version today, updating to v1.22.1 is advised to avoid any dependency changes. (go-version, cuda, etc)
    - The corresponding Boost version for this release is v1.6.3
- If you are on a Lotus v1.21.0-rcX version today, updating to v1.23.0 is advised to avoid any dependency changes. (go-version, cuda, etc). The v1.23.0 release will be published on Monday
    - The corresponding Boost version for this release is v1.7.0

---

Based on your team bandwidth/release/deployment flow, you can also consider a two phase update if you are currently running a v1.20.X-version

**Phase 1:** For the upgrade thats coming this Thursday (Apr 27th), simply update your nodes to v1.22.1

**Phase 2:** Post nv19 upgrade, review the Upgrade Warnings section in the v1.23.0 release notes, upgrade dependencies accordingly and upgrade to the v1.23.0 release.

**Reply by jennijuju** [reply]
if you have any questions, ask us in [#fil-lotus-help](https://filecoinproject.slack.com/archives/CPFTWMY7N).

**Comment by jennijuju** [comment]
cross posting from our [newsletter](https://lotus.filecoin.io/blog/this-week-in-lotus-week-19-2023/) 

<img width="674" alt="image" src="https://github.com/filecoin-project/lotus/assets/42981373/78b8a3f2-d31d-40cf-a297-7452a3e6b7b4">

**Comment by TippyFlitsUK** [comment]
This upgrade has now successfully passed.
